.github/workflows/main_ci.yml

```diff
@@ -24,8 +24,10 @@ jobs:
     - name: Install dependencies
       run: |
         python -m pip install --upgrade pip
-        pip install pyomo numpy scipy nose codecov coverage cplex==12.9
-        conda install mpi4py ipopt glpk
+        pip install pyomo numpy scipy nose codecov coverage cplex==12.9 sympy
+        conda install mpi4py metis ipopt glpk wntr networkx=2.3 --no-update-deps
```

Comment:
Why the hard-coded networkx and cplex versions?  Can you add a comment as to why this was pinned and what needs to be resolved to free the dependency.